client/src/components/molecules/EditUndertimeModal/EditTab.tsx

```diff
@@ -0,0 +1,688 @@
+
+const UndertimeTab: FC<Props> = ({ isOpen, closeModal }): JSX.Element => {
+  const router = useRouter()
+  const { leaveId } = router.query
+  const [managers, setManagers] = useState<UserType[]>([])
+  const [leaders, setLeaders] = useState<LeaderDetails[]>([])
+  const { handleProjectQuery, getLeadersQuery } = useProject()
+  const { handleAllUsersQuery, handleUserQuery } = useUserQuery()
+  const { handleLeaveTypeQuery, getSpecificLeaveQuery, handleUpdateLeaveMutation } = useLeave()
+  const { data: user } = handleUserQuery()
+  const leaveMutation = handleUpdateLeaveMutation(
+    user?.userById.id as number,
+    parseInt(router.query.year as string)
+  )
+  const { data: projects } = handleProjectQuery()
+  const { data: leaveTypes } = handleLeaveTypeQuery()
+  const { data: leadersList } = getLeadersQuery(undefined)
+  const { data: userLeave } = getSpecificLeaveQuery(Number(leaveId))
+  const { data: users, isSuccess: isUsersSuccess } = handleAllUsersQuery()
+
+  // Initial Data useStates
+  const [initialProjectNameValue, setInitialProjectNameValue] = useState<ReactSelectOption[]>([])
+
+  const [initialProjectLeaderValue, setInitialProjectLeaderValue] = useState<ReactSelectOption[]>(
+    []
+  )
+  const [initialLeaveTypeValue, setInitialLeaveTypeValue] = useState<ReactSelectOption | null>(null)
+  const [initialLeaveDateValue, setInitialLeaveDateValue] = useState<
+    Array<{ date: string; number_of_days_in_leave: ReactSelectOption; is_with_pay: boolean }>
+  >([])
+  const [initialManagerValue, setInitialManagerValue] = useState<ReactSelectOption | null>(null)
+  const [initialReasonValue, setInitialReasonValue] = useState<ReactSelectOption | null>(null)
+
+  const emptyReactSelectOption = {
+    label: '',
+    value: ''
+  }
+  const [isSaveDisable, setIsSaveDisable] = useState<boolean>(true)
+
+  // modify custom style control
+  customStyles.control = (provided: Record<string, unknown>, state: any): any => ({
+    ...provided,
+    boxShadow: 'none',
+    borderColor: 'none',
+    '&:hover': {
+      color: '#75c55e'
+    }
+  })
+
+  const {
+    control,
+    setValue,
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting }
+  } = useForm<NewLeaveFormValues>({
+    mode: 'onTouched',
+    resolver: yupResolver(NewLeaveSchema)
+  })
+
+  // This will handle form submit and save
+  const handleSave = async (data: NewLeaveFormValues): Promise<void> => {
+    // Other Project Name
+    const others = data.projects.map(
+      (project) => project.project_name.__isNew__ === true && project.project_name.value
+    )
+
+    await leaveMutation.mutateAsync(
+      {
+        userId: user?.userById.id as number,
+        leaveTypeId: parseInt(data.leave_type.value),
+        leaveId: Number(leaveId),
+        managerId: parseInt(data.manager.value),
+        reason: data.reason,
+        otherProject: others.filter((value) => value !== false).toString(),
+        leaveDates: data.leave_date.map((date) => {
```

Comment:
```suggestion
        leaveDates: data.leave_date.map((date) => ({
             leaveDate: date.date,
             isWithPay: date.is_with_pay,
             days: parseFloat(date.number_of_days_in_leave.value)
        })
```

Try to update it like this